### PR TITLE
Move driver_api to `infra`

### PR DESCRIPTION
**Pull request by sunce86**
# Description
Third part of the https://github.com/cowprotocol/services/pull/2188

Moves driver_api to a separate `solvers` module in `infra`.

Next up, moving the driver dtos into `infra`.

### CODE REVIEW

File: crates/autopilot/src/infra/solvers/mod.rs [hunk 1]
```diff
@@ -72,11 +77,29 @@ impl Driver {
...
+    let mut bytes = Vec::new();
+    while let Some(chunk) = response.chunk().await? {
+        let slice: &[u8] = &chunk;
+        if bytes.len() + slice.len() > limit {
+            return Err(anyhow!("size limit exceeded"));
+        }
+        bytes.extend_from_slice(slice);
+    }
+    Ok(bytes)
+}
```
Comment (sunce86) [inline crates/autopilot/src/infra/solvers/mod.rs:105]: copied from https://github.com/cowprotocol/services/blob/main/crates/shared/src/http_client.rs#L81 to break the dependency on `shared` crate.

File: crates/autopilot/src/run_loop.rs [hunk 2]
```diff
@@ -39,9 +38,9 @@ use {
 pub struct RunLoop {
     pub eth: infra::Ethereum,
     pub persistence: infra::Persistence,
+    pub drivers: Vec<infra::Driver>,
```
Comment (sunce86) [inline crates/autopilot/src/run_loop.rs:41]: should i rename this to `solvers: Vec<infra::Solver`? Do we want to refer to driver as `Solver` in the future?

File: crates/autopilot/src/util/mod.rs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+pub fn join(url: &Url, mut path: &str) -> Url {
+    let mut url = url.to_string();
+    while url.ends_with('/') {
+        url.pop();
+    }
+    while path.starts_with('/') {
+        path = &path[1..]
+    }
+    Url::parse(&format!("{url}/{path}")).unwrap()
+}
```
Comment (sunce86) [inline crates/autopilot/src/util/mod.rs:14]: copied from https://github.com/cowprotocol/services/blob/main/crates/shared/src/url.rs#L5 to break the dependency to `shared`. This is already done for `solvers` crate also: https://github.com/cowprotocol/services/blob/main/crates/solvers/src/util/url.rs

File: crates/autopilot/src/run_loop.rs (hunk 2 above)
Comment (MartinquaXD) [reply crates/autopilot/src/run_loop.rs:41]: Renaming makes sense to me. But I think I would only do it after the `autopilot` was completely DDD-ified.

**Review by MartinquaXD — APPROVED** [review]

File: crates/autopilot/src/run_loop.rs (hunk 2 above)
Comment (fleupold) [reply crates/autopilot/src/run_loop.rs:41]: I'd also support this renaming

**Review by fleupold — APPROVED** [review]

File: crates/autopilot/src/util/mod.rs (hunk 3 above)
Comment (sunce86) [reply crates/autopilot/src/util/mod.rs:14]: https://github.com/cowprotocol/services/issues/2285

File: crates/autopilot/src/run_loop.rs (hunk 2 above)
Comment (sunce86) [reply crates/autopilot/src/run_loop.rs:41]: Ok, let's keep the renaming for later. Anyway it will be trivial.

**Pull request merged**